### DISCUSSION TITLE
Daikin pinout information?

### DISCUSSION THREAD

**Discussion by matteius** [Arnold-n/P1P2MQTT#50] [Q&A] [answered]
Howdy -- I found this project in my own initial investigation towards combing a raspberry pi pico w with my Daikin heat pumps.  I have two CTXG12QVJUS heads, and here are some pictures of the connectors and my thoughts about them.  Please disregard the dust, I did clean the unit after taking this photos.

![image](https://user-images.githubusercontent.com/479892/209685247-5bc4ced9-5a1f-4e17-85f0-24b975dcb497.png)
Here we have the main power and two black connectors.   I am not sure what the longer connection is, but I suspect that the 4 wire connector to a 5-pin (one is unused) is some kind of P1,P2 Power and Ground but I am not sure if this is accurate, or if true what the exact pinout is.  I suspect that the right two pins are power and ground (not sure the order), but any insights would be helpful.

There is also this multicolor cable which I believe is a thermostat cable and not going to provide the functionality I am looking for that communicating with the P1P2 modbus would provide.
![image](https://user-images.githubusercontent.com/479892/209685653-030ca374-6ca4-4ae1-8066-4988d749f8c6.png)

I am equipped with an Eagle license and just had my first board fabbed for a different project that came out nice.  This is the next most important project I could apply my side electronic skills to.  I found a micropython modbus library https://github.com/alejoseb/Modbus-PI-Pico-FreeRTOS and plan to grok this code at some point to work on a micropython version.  However I am currently blocked on making 100% sure I know what I am connecting up to.  Also would be interesting to know what the power coming off that connection is, because if it were 5V I would love to power the pico circuit with it as well.

**Comment by gmcmicken** [comment, marked as answer]
It looks like that unit uses s21 protocol, which is different from the one covered by this library.

**Comment by Arnold-n** [comment]
Indeed, this looks like a S21 serial connector.
